spec/helpers/perf/drivers/terraform.lua

```diff
@@ -237,19 +263,31 @@ function _M:start_kong(version, kong_conf)
 
   local download_path
   if version:sub(1, 1) == "2" then
-    download_path = "gateway-2.x-ubuntu-focal"
+    if version:match("rc") or version:match("beta") then
+      download_path = "https://download-stage.konghq.com/gateway-2.x-ubuntu-focal/pool/all/k/kong/kong_" ..
+                      version .. "_amd64.deb"
+    else
+      download_path = "https://download.konghq.com/gateway-2.x-ubuntu-focal/pool/all/k/kong/kong_" ..
+                      version .. "_amd64.deb"
+    end
   else
     error("Unknown download location for Kong version " .. version)
   end
 
   local ok, err = execute_batch(self, self.kong_ip, {
     "echo > " .. KONG_ERROR_LOG_PATH,
     "sudo id",
+    -- set cpu scheduler to performance, it should lock cpufreq to static freq
     "echo performance | sudo tee /sys/devices/system/cpu/cpu*/cpufreq/scaling_governor",
-    "dpkg -l kong && (sudo kong stop; sudo dpkg -r kong) || true", -- stop and remove kong if installed
+    -- increase outgoing port range to avoid 99: Cannot assign requested address
+    "sysctl net.ipv4.ip_local_port_range='10240 65535'",
+    -- stop and remove kong if installed
+    "dpkg -l kong && (sudo kong stop; sudo dpkg -r kong) || true",
+    -- have to do the pkill sometimes, why?
```

Comment:
Perhaps `kong stop` sends the signal, but kong is allowed to "linger" for a while. Perhaps there's not enough CPUs on the CI to run the OS to actually remove the pidfile, or something along those lines.

Possible alternative to doing the pkill might be waiting for the pidfile to be removed (or for `pgrep nginx` to return empty)

Reply:
yeah i think i can adjust based on pid, let me test on it.